apiserver/facades/controller/instancepoller/merge.go

```diff
@@ -40,12 +46,14 @@ func (o *mergeMachineLinkLayerOp) Build(_ int) ([]txn.Op, error) {
 	// If the machine agent has not yet populated any link-layer devices,
 	// then we do nothing here. We have already set addresses directly on the
 	// machine document, so the incoming provider-sourced addresses are usable.
-	// For now we ensure that the instance poller only adds device information
+	// For now we ensure that the instance-poller only adds device information
 	// that the machine agent is unaware of.
 	if len(o.ExistingDevices()) == 0 {
 		return nil, jujutxn.ErrNoOperations
 	}
 
+	o.normaliseIncoming()
```

Comment:
I chose this technique of massaging the input so as to avoid overriding all the filtering/matching/accounting implemented in the base class shared by this model operation and the one in `networkingcommon`.

It also means we can have a single point of responsibility for handling the various policy changes we might implement, some of which are suggested in the patch description.